api/src/main/java/jakarta/data/metamodel/Attribute.java

```diff
@@ -25,111 +25,93 @@
+     * <p>Used by the Jakarta Data provider to initialize this {@code Attribute} with implementation.
+     * When initializing a subtype of {@code Attribute} with this method, the Jakarta Data provider
+     * must supply an instance of the subtype.</p>
      *
-     * @return a request for a descending sort on the entity attribute.
-     * @throws MappingException if an entity attribute with this name does not exist or
-     *                          if no Jakarta Data provider provides a repository for the entity type.
-     */
-    public final Sort desc() {
-        return attrInfo().desc();
-    }
-
-    /**
-     * Obtain a request for a descending, case insensitive {@link Sort} based on the entity attribute.
+     * <p>The default value of <code>false</code> is appropriate for instances that are
+     * pre-initialized, such as those that are provided by a Jakarta Data provider
+     * or are generated in advance with an annotation processor.</p>
      *
-     * @return a request for a descending, case insensitive sort on the entity attribute.
-     * @throws MappingException if an entity attribute with this name does not exist or
-     *                          if no Jakarta Data provider provides a repository for the entity type.
+     * @param attr attribute implementation provided by the Jakarta Data provider.
+     * @return true if initialization was successful; false if already initialized.
      */
-    public final Sort descIgnoreCase() {
-        return attrInfo().descIgnoreCase();
+    default boolean init(Attribute attr) {
+        return false;
     }
 
     /**
      * <p>Obtains a new instance for the Jakarta Data provider to initialize.</p>
      *
      * <p>The Jakarta Data provider automatically initializes the instance when
      * used as the value of a {@code public}, {@code static}, {@code final} field
-     * of a class that is annotated with {@link StaticMetamodel},
+     * of a class or interface that is annotated with {@link StaticMetamodel},
      * where the field name matches the name of an entity attribute.
-     * For example, a {@code Vehicle} entity with attributes {@code vin}, {@code make},
-     * {@code model}, and {@code year} could have the following static metamodel,</p>
+     * For example, a {@code Person} entity with attributes {@code id}, {@code emails},
+     * {@code phoneNumbers}, and {@code name} could have the following static metamodel,</p>
      *
      * <pre>
-     * &#64;StaticMetamodel(Vehicle.class)
-     * public class Vehicle_ {
-     *     public static final Attribute vin = Attribute.get();
-     *     public static final Attribute make = Attribute.get();
-     *     public static final Attribute model = Attribute.get();
-     *     public static final Attribute year = Attribute.get();
+     * &#64;StaticMetamodel(Person.class)
+     * public interface Person {
+     *     SortableAttribute id = SortableAttribute.get();
```

Comment:
We need to be open to both runtime and build time.
Once you are not defining it as final, you can keep it open as Jakarta Persistence does.

Reply:
> We need to be open to both runtime and build time. Once you are not defining it as final, you can keep it open as Jakarta Persistence does.
> 
> Ref: https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1#a6961

No, it **is** defined as `final` because `final` is implicit for attributes of interfaces.  That hasn't changed.  You need to explain what you mean that this is not open to build time.  We still have a TCK test that covers build time, and that test still compiles and passes with this approach, so your comment there makes no sense to me.

Reply:
I will create a prototype for both scenarios: build time and runtime, thus, we can discuss this in check it in action.

Reply:
> I will create a prototype for both scenarios: build time and runtime, thus, we can discuss this in check it in action.

Yes, go ahead. But please understand that the TCK already has a prototype of both build time and run time styles.  And I've already updated an implementation accordingly and have seen both scenarios working in the TCK with the changes under this pull, so it has been verified in action.  Hopefully it helps when you try it out as well.